shared/route-tree/index.js

```diff
@@ -99,7 +99,7 @@ type PathIterable<X> = I.IndexedSeq<X> | I.List<X> | Array<X>
 export type Path = PathIterable<string>
 export type PropsPath<P> = PathIterable<string | {selected: string | null, props: P}>
 type PathParam<P> = [] | Path | PropsPath<P>  // Flow doesn't accept Path as a subtype of PropsPath, so be explicit here.
-type PathSetSpec<P> = I.IndexedIterable<{type: 'next' | 'navigate', next: string | null, props?: P}>
+type PathSetSpec<P> = I.IndexedIterable<{type: 'traverse' | 'navigate', next: string | null, props?: P}>
```

Comment:
Thinking about this again, I still don't like traverse vs next and navigate. They are all the same word.

how about we call one `navigateAndSelecting` and the other `navigateWithoutSelecting`